app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        $this->projectpath = $this->basepath . DIRECTORY_SEPARATOR . $this->projectname;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem;
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
+            if ($fs->deleteDirectory($this->projectpath)) {
+                $this->info("I removed the following directory: {$this->projectpath}");
+
+                return true;
+            } else {
+                $this->error("I was unable to remove the '{$this->projectpath}' directory so I must exit.");
+
+                return false;
+            }
+        } else {
+            $this->error("You have chosen to not remove the '{$this->projectpath}' directory so I must exit.");
+
+            return false;
+        }
+    }
+
+    protected function replaceEnvVariables()
+    {
+        // @todo make this a configuration option
+        $changes = [
+            'DB_DATABASE' => $this->projectname,
+            'DB_USERNAME' => 'root',
+            'DB_PASSWORD' => '',
+            'APP_URL'     => $this->projecturl,
+        ];
+
+        return $this->updateDotEnv($changes);
+    }
+
+    protected function doNodeOrYarn()
+    {
+        $command = null;
+        if ($this->hasTool('yarn')) {
+            $command = 'yarn';
+        } elseif ($this->hasTool('npm')) {
+            $command = 'npm install';
+        }
+
+        if (empty($command)) {
+            $this->error('Either yarn or npm are required');
+
+            return false;
+        }
+
+        $this->info("Executing $command now; in {$this->projectpath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+        $process->start();
+
+        $iterator = $process->getIterator($process::ITER_SKIP_ERR | $process::ITER_KEEP_OUTPUT);
+        foreach ($iterator as $data) {
+            $this->line($data);
+        }
+    }
+
+    protected function chooseDatabase($dbtype)
+    {
+        if (! in_array($dbtype, $this->dbtypes)) {
+            $this->info("You passed in '--createdb '{$dbtype}' but I do not understand");
+            $type = $this->anticipate('What type of database would you like to install? Options are '.implode(' or ', $this->dbtypes) . '.', $this->dbtypes, $this->dbtypes[0]);
```

Comment:
please add spaces around all concatenators (`.`)

Reply:
also, different structure allows us to only hit this conditional if there's anything wrong:

```php
    protected function chooseDatabase($type)
    {
        if (! in_array($type, $this->dbtypes)) {
            $this->info("You passed in '--createdb '{$type}' but I do not understand");
            $type = $this->anticipate('What type of database would you like to install? Options are ' . implode(' or ', $this->dbtypes) . '.', $this->dbtypes, $this->dbtypes[0]);
        }
```

Finally, I'd probably split that `anticipate` line out:

```php
            $type = $this->anticipate(
                'What type of database would you like to install? Options are ' . implode(' or ', $this->dbtypes) . '.', 
                $this->dbtypes,
                reset($this->dbtypes)
            );
```

.... oops, just read on. Here's the whole thing:

```php
    protected function chooseDatabase($type)
    {
        if (! in_array($type, $this->dbtypes)) {
            $this->info("You passed in '--createdb '{$type}' but I do not understand");

            $type = $this->anticipate(
                'What type of database would you like to install? Options are ' . implode(' or ', $this->dbtypes) . '.', 
                $this->dbtypes,
                reset($this->dbtypes)
            );

            if (! in_array($type, $this->dbtypes)) {
                return $this->alert("Sorry, but {$type} is not a valid database type. Skipping.");
            }
        }

        $this->info("Creating a new {$type} database");

        if ($this->createDatabase($type)) {
            $this->info("Executing 'php artisan migrate:fresh'");
            $this->migrateFresh();
        }
    }
```

Reply:
Also.. we could instead use `choice`? 

```php
$name = $this->choice('What is your name?', ['Taylor', 'Dayle'], $default);
```